plugins/module_utils/nd_argument_specs.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+def ntp_server_spec():
+    return dict(
+        ntp_host=dict(type="str", required=True),
+        ntp_key_id=dict(type="int", no_log=False),
+        preferred=dict(type="bool", default=False),
+    )
+
+
+def ntp_keys_spec():
+    return dict(
+        ntp_key_id=dict(type="int", required=True, no_log=False),
+        ntp_key=dict(type="str", required=True, no_log=False),
```

Comment:
We might need this to be obfuscated:
```suggestion
        ntp_key=dict(type="str", required=True, no_log=True),
```